docs/DeveloperGuide.md

```diff
@@ -5,25 +5,106 @@
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
 ## Design & implementation
-
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Overview
+![Overview.png](Overview.png)
+Given above is a overview of how each classes interact with one another in our software.
+
+### Save feature
```

Comment:
I really like your diagram for this! very clear and straightforward + easy to understand!